### PR TITLE
Improve PRAGMA integrity_check(check_nav_class_ids) performance

### PR DESCRIPTION
**Pull request by simasjar**

### CODE REVIEW

File: iModelCore/ECDb/ECDb/IntegrityChecker.cpp [hunk 1]
```diff
@@ -731,30 +752,63 @@ DbResult IntegrityChecker::CheckNavClassIds(std::function<bool(ECInstanceId, Utf
...
+			std::string derivedIds = "";
+			bool isFirst = true;
+			for (auto & derivedNavClassId : derivedNavClassIds) {
+				if (!isFirst) {
+					derivedIds += ", ";
+				}
+				isFirst = false;
+				derivedIds += std::to_string(derivedNavClassId.GetValue());
+			}
+			std::string incorrectClassIdsForTableQuery = SqlPrintfString("SELECT DISTINCT %s.RelECClassId FROM %s WHERE %s.RelECClassId IS NOT NULL AND %s.RelECClassId NOT IN (%s)",
```
[the remark this answers is not in the document]
Comment (khanaffan) [reply iModelCore/ECDb/ECDb/IntegrityChecker.cpp:769]: i go one step further you can do it in single SQL.

File: iModelCore/ECDb/ECDb/IntegrityChecker.cpp [hunk 2]
```diff
@@ -7,6 +7,27 @@
 USING_NAMESPACE_BENTLEY_EC
 USING_NAMESPACE_BENTLEY_SQLITE
 BEGIN_BENTLEY_SQLITE_EC_NAMESPACE
+//---------------------------------------------------------------------------------------
+// @bsimethod
+//+---------------+---------------+---------------+---------------+---------------+------
+DbResult IntegrityChecker::GetDerivedNavigationIds(Utf8CP id, std::vector<ECClassId>& derivedECClassIds) {
+	auto sql = SqlPrintfString("SELECT [ClassId] FROM [main].[ec_cache_ClassHierarchy] WHERE [BaseClassId] = %s", id);
```
Comment (khanaffan) [inline iModelCore/ECDb/ECDb/IntegrityChecker.cpp:14]: Use cached statement with paramters.

File: iModelCore/ECDb/ECDb/IntegrityChecker.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (khanaffan) [reply iModelCore/ECDb/ECDb/IntegrityChecker.cpp:769]: Another thing it seem my code already looked for this case but may be bug there.

I used ec_instanceOf which check if a class is subclass of another class or  not.

`ec_instanceof(s.%s.RelECClassId, %s) = 0)`

[the remark this answers is not in the document]
Comment (simasjar) [reply iModelCore/ECDb/ECDb/IntegrityChecker.cpp]: Changed to `WHERE %s.RelECClassId IS NOT NULL AND %s.RelECClassId NOT IN (SELECT SourceECInstanceId FROM ECDbMeta.ClassHasAllBaseClasses WHERE TargetECInstanceId = %s)`

File: iModelCore/ECDb/ECDb/IntegrityChecker.cpp (hunk 2 above)
Comment (simasjar) [reply iModelCore/ECDb/ECDb/IntegrityChecker.cpp]: I have removed this function

**Review by khanaffan — APPROVED** [review]

**Pull request merged**